### ISSUE TITLE
time "yyyy-MM-dd" parse failed.

### ISSUE THREAD

**Issue by slangmgh** [nim-lang/Nim#10142]
The following code fail sometime:
```nim
import times
echo "1990-12-19".parse("yyyy-MM-dd")
```

The code output:
```nim
times.nim(2264)          t3
times.nim(2268)          parse
times.nim(2238)          parse
times.nim(1617)          raiseParseException
Error: unhandled exception: Failed to parse '1990-12-19' with format 'yyyy-MM-dd'. Failed on pattern 'yyyy' [TimeParseError]
Error: execution of an external program failed: 'D:\Tool\Nim\Stock\t3.exe '
```

The following code in times.parseInt always return 0:
```nim
      if b >= (low(int) + c) div 10:
        b = b * 10 - c
      else:
        return 0
```

**Comment by slangmgh** [comment]
The code is fine when compile with "-d:release".
I use vc2013.

**Comment by Araq** [comment]
Works for me with the Gnu C compiler and also with ``--cc:vcc``.

**Comment by slangmgh** [comment]
Fixed after the #10377 fixed.
